### DISCUSSION TITLE
Control SUMO vehicles mobility depending on V2X communication

### DISCUSSION THREAD

**Discussion by abdullah-traboulsi** [eclipse/mosaic#32] [Q&A] [answered]
HI,

I would like to ask if it is possible to control vehicles in SUMO depending on the received network communication from NS-3 or Omnet++?
I read that you can import SUMO configuration to MOSAIC, but what about online integration simulation between communication and mobility?

Thank you

**Comment by kschrab** [comment, marked as answer]
Sure, that's exactly what MOSAIC was made for in the first place. Next to coupling SUMO with NS-3 or OMNeT++ it provides the possibility to adjust the behavior of vehicles based on specific events. To achieve this, you would need to implement an application model for the Application Simulator. This application could send V2X messages which other vehicles would receive. When receiving a message, you could then adjust the behavior of the vehicle (e.g. change speed, change lane, change route, stop, resume, and more). For first testing you could use already existing applications, but writing your own is just as easy. You can read a lot more on that in our [online documentation](https://eclipse.org/mosaic/docs) and our [tutorials](https://eclipse.org/mosaic/tutorials).

Please note that it is currently not possible to include already existing SUMO configurations directly to MOSAIC without certain limitations. But it is very simple to create fresh mobility scenarios from OpenStreetMap data instead (again following our documentation).
